src/site/SearchResultSet.php

```diff
@@ -0,0 +1,60 @@
+<?hh // strict
+
+namespace HHVM\UserDocumentation;
+
+final class SearchResultSet {
+    private Map<string, string> $classes;
+    private Map<string, string> $functions;
+    private Map<string, string> $interfaces;
+    private Map<string, string> $hhvm_guides;
+    private Map<string, string> $hack_guides;
+
+    public function __construct() {
+        $this->classes = Map {};
+        $this->functions = Map {};
+        $this->interfaces = Map {};
+        $this->hhvm_guides = Map {};
+        $this->hack_guides = Map {};
+    }
+
+    public function addAPIResult(string $type, string $name): void {
+        if ($type === "class") {
+            $this->classes[$name] = sprintf("/hack/reference/class/%s/", $name);
+        }
+        if ($type === "function") {
+            $this->functions[$name] = sprintf("/hack/reference/function/%s/", $name);
+        }
+        if ($type === "interface") {
+            $this->interfaces[$name] = sprintf("/hack/reference/interface/%s/", $name);
+        }
```

Comment:
It works fine because the pathing uses `.` instead of `\`, for example: `/hack/reference/function/HH.Asio.null/`